.github/integration/tests/20_upload.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+set -e
+# inferred from access_key in testing/s3cmd.conf
+user=test_dummy.org
+
+
+# Function checking that a file was encrypted
+function check_encypted_file() {
+
+    for k in $1
+    do
+        output=$(head -c 8 "$k")
+
+        if [[ "$output" = "crypt4gh"  ]]; then
+            echo "Encrypted data file"
+        else
+            echo "Failed to encrypt file"
+            exit 1
+        fi
+    done
+}
+
+# Function checking that a file was uploaded to the S3 backend
+function check_uploaded_file() {
+    # TODO if emtpy, this will fail silently
+    output=$(s3cmd -c testing/directS3 ls s3://"$1" | grep -q "$2")
+    if $output ; then
+        echo "Uploaded encrypted file to s3 backend"
+    else
+        echo "Failed to upload file to s3 backend"
+        exit 1
+    fi
+}
+
+
+# Upload a specific file and check it
+./sda-cli -config testing/s3cmd.conf upload data_file.c4gh
+check_uploaded_file "test/$user/data_file.c4gh" data_file.c4gh
+
+
+# Try to upload a file twice with the --force-overwrite flag
+output=$(./sda-cli -config testing/s3cmd.conf upload --force-overwrite data_file.c4gh)
+
+
+# Test recursive folder upload
+for k in data_file.c4gh data_file1.c4gh
+do
+    # Upload and check file
+    ./sda-cli -config testing/s3cmd.conf upload --force-overwrite "data_files_enc/$k"
+    check_uploaded_file "test/$user/$k" "$k"
+done
```

Comment:
This is not a recursive upload as per comment.

Reply:
Fixed in 38a8ef7.